### PR TITLE
add no-cb-test rule (fixes #2)

### LINKED ISSUE #2: `no-cb-test` rule
Prevent the use of `test.cb()`. Some projects/companies might want to prevent the use of callback tests in favor of only async functions or returning promises.

### PR DESCRIPTION
**Pull request by jfmengels**
Adds no-cb-test rule (fixes #2) :)

cc @sindresorhus @twada (@twada let us know if you want us to stop pinging you ;) )

### CODE REVIEW

File: docs/rules/no-cb-test.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Ensure no `test.cb()` is used
+
+Disallow the use of `test.cb()`.
```
Comment (jfmengels) [inline docs/rules/no-cb-test.md]: Couldn't think of a good description, but might be nice to have more explanations.

File: docs/rules/no-cb-test.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+## Pass
+
+```js
+const test = require('ava');
+
+test('some test', t => {
+	t.pass();
+});
```
Comment (sindresorhus) [inline docs/rules/no-cb-test.md]:
```js
test('some test', async t => {
	t.pass();
});
```

would be a better example as it's the alternative we recommend over `test.cb()`.

File: readme.md [hunk 3]
```diff
@@ -44,6 +45,7 @@ Configure it in `package.json`.
 
 The rules will only activate in test files.
 
+- [no-cb-test](docs/rules/no-cb-test.md) - Ensure no `test.cb()` is used.
```
Comment (sindresorhus) [inline readme.md]: > Ensure `test.cb()` is not used.

?

File: docs/rules/no-cb-test.md (hunk 1 above)
Comment (sindresorhus) [inline docs/rules/no-cb-test.md]: > We instead recommend using `test()` with an async function or returning a promise.

?

File: docs/rules/no-cb-test.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Ensure no `test.cb()` is used
+
+Disallow the use of `test.cb()`.
+
+
+## Fail
+
+```js
+const test = require('ava');
```
Comment (sindresorhus) [inline docs/rules/no-cb-test.md]: Use `import test from 'ava';`

File: docs/rules/no-cb-test.md (hunk 1 above)
Comment (jfmengels) [inline docs/rules/no-cb-test.md]: Sounds good! I disabled the rule by default though, as its not a rule that every codebase can afford to enforce.

Comment (sindresorhus) [inline docs/rules/no-cb-test.md]: Agreed

**Comment by jfmengels** [comment]
@sindresorhus All the recommendations sounded good, updated ;)

**Comment by sindresorhus** [comment]
🎉🍻

**Comment by jfmengels** [comment]
\o/

**Comment by jfmengels** [comment]
First merge not creating conflicts on the others too!

**Pull request merged**